src/ngrx/4-create-actions/create-actions.md

```diff
@@ -1,4 +1,64 @@
 @page learn-ngrx/create-actions Creating Actions
 @parent learn-ngrx 4
 
-@description Learn how to create NgRx actions.
\ No newline at end of file
+@description Learn how to create NgRx actions.
+
+## Overview
+
+TODO
+
+## Problem 1: Create Login Actions to Represent Login Events
+
+NgRx Schematics generated Actions with the following Action Types: `[Login] Load Logins`, `[Login] Load Logins Success` and `[Login] Load Logins Failure`.
+
+Our goal is to have a different set of Action instead:
+
+- `login` - dispatched when user submits username and password
+    - type: `[Login Page] Login`
+    - props: `{ username: string, password: string }`
+- `loginSuccess` - dispatched by an Effect when user has successfully logged in
+    - type: `[Login API] Login Success`
+    - props: `{ userId: string; username: string; token: string }`
+- `loginFailure` - dispatched by an Effect when user login attempt fails
+    - type: `[Login API] Login Failure`
+    - props: `{ errorMsg: string }`
+- `logout` - dispatched when user clicks on a logout button
+    - type: `[Chat Page] Logout`
+
+## P1: What You Need to Know
+
+Action Types in NgRx follow a string pattern: `[Source] Event`. Since Actions represent a unique event, an Action Type should be a unique identifier for that event. NgRx encourages Actions to be unique in order to help with application debugging, traceability and maintainability. [Here’s a great presentation about Action Best Practices](https://www.youtube.com/watch?v=JmnsEvoy-gY) from Mike Ryan, co-creator of NgRx.
+
+The code example below shows the usage of the createAction function as generated by NgRx Schematics:
+
+@sourceref ./login.actions-starter.ts
+
+`createAction` can take up to two arguments: the string `type` and a `config` function that represents additional metadata, usually referred as `props` in Redux Pattern.
+
+## P1: Solution
+
+TODO
+
+## Problem 2: Create Auth State Interface and Initial State
+
+Now that our Actions are defined, we need to set up the Auth State and its Initial State to enable to perform updates to the state later.
+
+We’ll define the following members of the Auth’s State interface:
+
+- `userId: string | null`;
+- `username: string | null`;
+- `token: string | null`;
+
+As for the `initialState`, members should start with `null` values.
```

Comment:
I've used null values but I don't know if this is the best approach, I'm curious to what your opinion is

Reply:
I like null since it hints that there is expected to be a value at some point. There's nothing wrong with using undefined, but I'd always go with null if there's freedom to choose